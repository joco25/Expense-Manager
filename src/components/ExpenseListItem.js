import React from 'react'
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';



const ExpenseListItem= ({ id, description, amount,createdAt})=>(
    <div>
        <Link to={`/edit/${id}`}><h3>{description}</h3></Link>
        <span>
            {numeral(amount / 100).format('$0,0.00')} 
                - 
            {moment(createdAt).format('MMM Do, YYYY')}</span>
    </div>
)


export default ExpenseListItem