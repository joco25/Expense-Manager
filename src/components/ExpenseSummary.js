import React from 'react';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';
import {connect} from 'react-redux';

export const ExpenseSummary=({expenseCount,expenseTotal})=> {
    const expenseWord = expenseCount === 1 ? 'expense': 'expenses';
    const formattedExpensesTotal = numeral(expenseTotal/ 100).format('$0,0.00');
    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
        </div>
    )
};

const mapStateToProps=(state)=>{
    const visibleExpenses= selectExpenses(state.expenses,state.filters);

    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: selectExpensesTotal(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpenseSummary);