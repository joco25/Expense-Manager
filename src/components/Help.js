import React from "react";
const Help = () => (
    <div >
        Get help
    </div>
)

export default Help