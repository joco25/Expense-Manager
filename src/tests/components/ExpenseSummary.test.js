import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseSummary} from '../../components/ExpenseSummary';

test('should display singular for one expense',()=>{
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expenseTotal={3000}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should display singular for one expense',()=>{
    const wrapper = shallow(<ExpenseSummary expenseCount={14} expenseTotal={35748000}/>);
    expect(wrapper).toMatchSnapshot();
});