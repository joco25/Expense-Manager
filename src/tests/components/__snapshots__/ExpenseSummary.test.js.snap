// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display singular for one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $30.00
  </h1>
</div>
`;

exports[`should display singular for one expense 2`] = `
<div>
  <h1>
    Viewing 
    14
     
    expenses
     totalling 
    $357,480.00
  </h1>
</div>
`;
